jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1111,15 +1106,22 @@ private void parsedTrailer()
 
     private long convertContentLength(String valueString)
     {
-        try
-        {
-            return Long.parseLong(valueString);
-        }
-        catch (NumberFormatException e)
+        if (valueString == null || valueString.length() == 0)
+            throw new BadMessageException("Invalid Content-Length Value", new NumberFormatException());
+
+        long value = 0;
+        int length = valueString.length();
+
+
+        for (int i = 0; i < length; i++)
```

Comment:
It feels like the logic of this loop should be moved to `org.eclipse.jetty.util.MathUtils`.